scripts/endToEndExtraction/remove-testcases.py

```diff
@@ -61,51 +61,51 @@ def get_checks(content, sol_file_path):
             constructors.append(line)
         if line.startswith("ABI_CHECK") or line.startswith("BOOST_REQUIRE"):
             checks.append(line)
-    sol_file = open(sol_file_path, "r")
-    sol_constructors = []
-    sol_checks = []
-    inside_expectations = False
-    for line in sol_file.readlines():
-        if line.startswith("// constructor()"):
-            sol_constructors.append(line)
-        elif inside_expectations and line.startswith("// "):
-            sol_checks.append(line)
-        if line.startswith("// ----"):
-            inside_expectations = True
-    sol_file.close()
-    if len(constructors) == len(sol_constructors) == 1:
-        checks.insert(0, constructors[0])
-        sol_checks.insert(0, sol_constructors[0])
-    return checks, sol_checks
+    with open(sol_file_path, "r") as sol_file:
+        sol_constructors = []
+        sol_checks = []
+        inside_expectations = False
+        for line in sol_file.readlines():
+            if line.startswith("// constructor()"):
+                sol_constructors.append(line)
+            elif inside_expectations and line.startswith("// "):
+                sol_checks.append(line)
+            if line.startswith("// ----"):
+                inside_expectations = True
+        sol_file.close()
```

Comment:
This is no longer needed if you're using `with`. Also, the `with` block should really end here.